### PR TITLE
core: arbitrary message payload size; Fix #48

### LINKED ISSUE #48: Messages payload are limited to 4000 characters and throw exception
When inserting messages to hermes which are too long this throws a DB exception

`DatabaseException: (raised as a result of Query-invoked autoflush; consider using a session.no_autoflush block if this flush is occurring prematurely) (cx_Oracle.DatabaseError) ORA-01461: can bind a LONG value only for insert into a LONG column [SQL: 'INSERT INTO atlas_rucio.messages (id, event_type, payload, updated_at, created_at) VALUES (:id, :event_type, :payload, :updated_at, :created_at)'] [parameters: {'id': , 'created_at': datetime.datetime(2017, 5, 18, 14, 26, 32, 316246), 'event_type': 'email', 'updated_at': datetime.datetime(2017, 5, 18, 14, 26, 32, 316240), 'payload': '{"body": "A new rule has been requested for approval in Rucio.\\n\\nRule description:\\n\\n ID: 89593c3ea1e449cea3b2dbbd81033b06\\ ... (3749 characters truncated) ... TED MESSAG\\n", "to": ["damian.alvarez.piqueras@cern.ch"], "subject": "[RUCIO] Request to approve replication rule 89593c3ea1e449cea3b2dbbd81033b06"}'}]`

We have to look into if we can update the DB to allow longer messages, but in anyway, hermes should protect against this.

_Migrated from JIRA [RUCIO-2662](https://its.cern.ch/jira/browse/RUCIO-2662)_

### PR DESCRIPTION
**Pull request by mlassnig**
core: arbitrary message payload size; Fix #48

### CODE REVIEW

File: lib/rucio/core/message.py [hunk 1]
```diff
@@ -32,19 +40,32 @@ def add_message(event_type, payload, session=None):
...
+
+    # If we allow SQL context switches and messages are larger than default 4000
+    # then write magic character ^K into NOT NULL payload column
+    nolimit = config.get('messages', 'payload_nolimit', default=False, session=session)
+    if nolimit and len(payload) > 4000:
+        new_message = Message(event_type=event_type, payload='nolimit', payload_nolimit=payload)
+    else:
+        try:
+            new_message = Message(event_type=event_type, payload=payload)
+        except DatabaseError, e:
```
Comment (cserf) [inline lib/rucio/core/message.py]: Not py3k compatible

File: lib/rucio/core/message.py [hunk 2]
```diff
@@ -101,11 +122,22 @@ def retrieve_messages(bulk=1000, thread=None, total_threads=None, event_type=Non
...
+                nolimit_query = session.query(Message.payload_nolimit).filter(Message.id == id).one()[0]
+                message['payload'] = json.loads(str(nolimit_query))
+            else:
+                message['payload'] = json.loads(str(payload))
+
+            messages.append(message)
+
         return messages
 
     except IntegrityError, e:
```
Comment (cserf) [inline lib/rucio/core/message.py]: Not py3k compatible

File: lib/rucio/db/sqla/migrate_repo/versions/90f47792bb76_add_clob_payload_to_messages.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Revises: 1f46c5f240ac
+# Creation Date: 2019-01-21 15:35:22.732801
+
+from alembic.op import add_column, drop_column
+
+import sqlalchemy as sa
+
+# revision identifiers used by alembic
+revision = '90f47792bb76'       # pylint: disable=invalid-name
+down_revision = '1f46c5f240ac'  # pylint: disable=invalid-name
```
Comment (cserf) [inline lib/rucio/db/sqla/migrate_repo/versions/90f47792bb76_add_clob_payload_to_messages.py]: Be careful that there is another PR with alembic changes. We might need to change the down_revision if the other PR is merged first

Comment (bari12) [reply lib/rucio/db/sqla/migrate_repo/versions/90f47792bb76_add_clob_payload_to_messages.py]: The down_revision I am adapting in the merge; This should be fine;

**Review by cserf — DISMISSED** [review]

**Review by bari12 — APPROVED** [review]
